encoding/protoavro/marshal.go

```diff
@@ -28,11 +34,12 @@ func NewMarshaler(descriptor protoreflect.MessageDescriptor, writer io.Writer) (
 	if err != nil {
 		return nil, fmt.Errorf("new ocf writer: %w", err)
 	}
-	return &Marshaler{w: w, desc: descriptor}, nil
+	return &Marshaler{w: w, desc: descriptor, SchemaOptions: o}, nil
 }
 
 // Marshaler encodes and writes Avro binary encoded messages.
 type Marshaler struct {
+	SchemaOptions
```

Comment:
Use a regular field for this, we don't want to export the `SchemaOptions` on an already created `Marshaler`.
Ie `opts SchemaOptions`